controllers/configmap_controller.go

```diff
@@ -330,9 +330,14 @@ func (r *ConfigMapReconciler) deconfigureInstances(instances []*instance.Info, n
 }
 
 // hasAssociatedInstance returns true if any of the given addresses is associated with any instance in the given slice.
-// The instance's network address must be a valid IPv4 address or resolve to one.
+// The instance's network address must be a valid internal IPv4 address or resolve to one.
 func hasAssociatedInstance(nodeAddresses []core.NodeAddress, instances []*instance.Info) (bool, error) {
 	for _, nodeAddress := range nodeAddresses {
+		// check node's address type
+		if nodeAddress.Type == core.NodeExternalIP || nodeAddress.Type == core.NodeExternalDNS {
+			// skip external addresses
+			continue
+		}
```

Comment:
I dont see why we should skip external addresses completely, it seems like we should instead not fail on reverse lookup failure on these lines below, and continue trying other addresses
```
		if parseAddr := net.ParseIP(nodeAddress.Address); parseAddr != nil {
			dnsAddresses, err := net.LookupAddr(nodeAddress.Address)
			if err != nil {
				return false, errors.Wrapf(err, "failed to lookup DNS for IP %s", nodeAddress.Address)
			}
			```

Reply:
Makes sense, adjusted in https://github.com/openshift/windows-machine-config-operator/pull/1045/commits/95d053bbbeb62c300887fb9828a454679aa540f5